mysys/charset.c

```diff
@@ -1107,7 +1107,7 @@ size_t escape_string_for_mysql(CHARSET_INFO *charset_info,
   const char *to_start= to;
   const char *end, *to_end=to_start + (to_length ? to_length-1 : 2*length);
   my_bool overflow= FALSE;
-  for (end= from + length; from < end; from++)
+  for (end= (length == 0)? from : from + length; from < end; from++)
```

Comment:
Is this supposed to fix "Attempting to offset a null pointer by zero"? It would be better to check if `from` is `NULL` or `length` is `0` and return surely?

Reply:
This was reverted as it broke spider tests.